docs/architecture.md

```diff
@@ -65,10 +65,22 @@ charm-microk8s:                     # Root directory
 - docs:
   - architecture.md                 # Document architecture decisions for the charm
   - development.md                  # Getting started with developing the charm and running tests
+- lib:
```

Comment:
I feel that this list of files is doomed to be stale. Maybe we should stop at the level of directories and not files.